### ISSUE TITLE
Vulkan: 2D msaa does not AA clipped children

### ISSUE THREAD

**Issue by insomniacUNDERSCORElemon** [godotengine/godot#66005]
### Godot version

4.0 Beta 1

### System information

Shouldn't be system specific. Linux, Vulkan, 1050 Ti nvidia driver v515.65.01

### Issue description

The MSAA, it does nothing for the clipped goggles. Unclipped is on the left for comparison, and the zoomed squares to better show the AA:

![mrp_screenshot](https://user-images.githubusercontent.com/4397725/190875501-c70cc146-aa8f-4e0d-9b71-cefc13362dc2.png)

@Geometror 

The ghosting, darkening, and blending on the clipped version are unrelated to MSAA. See https://github.com/godotengine/godot/issues/49198

clip_children might be better done in some other technical way (one that does not block MSAA hopefully) but I don't know enough to say.

### Steps to reproduce

1. Set up 2D project
2. Enable 2D MSAA x8 in project settings
3. Create any canvasitem node (sprite or polygon)
4. Create a polygon (or line) node as a child of previous step, enable clip_children on said parent
5. Run scene, the parent will have AA but the child will not

Note on step 4: I imagine a sprite could have the same issue, but polygons are an easier way to create something that needs AA (and they need AA always, whereas sprites can be smoothed if you can use transparency).

### Minimal reproduction project

[clip_msaa_mrp.zip](https://github.com/godotengine/godot/files/9592571/clip_msaa_mrp.zip)

(screenshot above is what the MRP is)

**Comment by Calinou** [comment]
I wonder if 2D clipping is performed in a shader. If so, would alpha-to-coverage make it able to be antialiased?

**Comment by insomniacUNDERSCORElemon** [comment]
> I wonder if 2D clipping is performed in a shader. If so, would alpha-to-coverage make it able to be antialiased?

Yeah, that's what I was talking about right before the r.steps. Also you'd know more than I. If it makes clip-related nodes visually identical to their unclipped form (thus no limitations), that's what I want.

Krita has a feature like this (alpha inheritance), not sure how suitable their technique would be for Godot's real-time rendering though.
